engines/stark/ui/menu/diaryindex.cpp

```diff
@@ -102,6 +102,16 @@ void DiaryIndexScreen::open() {
 	}
 }
 
+void DiaryIndexScreen::onScreenChanged() {
+	for (uint i = 1; i < _widgets.size(); i++) {
+		// Force the texture to be reset by changing the color
+		if (_widgets[i]->containsText()) {
+			_widgets[i]->setTextColor(_textColorHovered);
+			_widgets[i]->setTextColor(_textColorDefault);
```

Comment:
Here, calling `setTextColor` twice to reset the texture, while a perfectly working solution feels like a hack. Now when changing `setTextColor`, we need to make sure it does not break screen resizing. It would be better to have a specialized call to reset the text visual. One that could be reused to reset other text visuals in the game where this problem occurs.

Reply:
Right this puts an extra dependency on ```setTextColor```... As said in the description, I don’t want to change too many codes, but now I guess opening a specialized interface on class ```VisualText``` is likely a must. I’ll commit this as soon as possible.